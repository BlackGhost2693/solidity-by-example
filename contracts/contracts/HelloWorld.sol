// compiler version must be greater than or equal to 0.5.16 and less than 0.6.0
pragma solidity ^0.5.16;

contract HelloWorld {
  string public greet = "Hello World!";
}
