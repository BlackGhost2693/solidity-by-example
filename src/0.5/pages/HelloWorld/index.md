`pragma` specifies the compiler version of Solidity.

```solidity
{{{HelloWorld}}}
```
