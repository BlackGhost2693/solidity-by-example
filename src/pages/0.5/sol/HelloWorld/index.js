import React from "react"
import Example from "../../../components/Example"
import html from "./index.html.js"
import { SOL_VERSION } from "../constants"

export default function (props) {
  return (
    <Example
      title="Hello World"
      version={SOL_VERSION}
      description="Hello World in Solidity"
      html={html}
    />
  )
}
