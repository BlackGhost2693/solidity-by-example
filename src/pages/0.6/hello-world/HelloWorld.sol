// SPDX-License-Identifier: MIT
// compiler version must be greater than or equal to 0.6.10 and less than 0.7.0
pragma solidity ^0.6.10;

contract HelloWorld {
    string public greet = "Hello World!";
}
