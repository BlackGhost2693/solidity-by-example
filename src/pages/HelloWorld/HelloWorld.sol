// compiler version must be greater than or equal to 0.5.11 and less than 0.6.0
pragma solidity ^0.5.11;

contract HelloWorld {
  string public greet = "Hello World!";
}
