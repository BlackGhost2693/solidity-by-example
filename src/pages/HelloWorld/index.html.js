const html = `<p><code>pragma</code> specifies the compiler version of Solidity.</p>
<pre><code class="language-solidity">// compiler version must be greater than or equal to 0.5.3 and less than 0.6.0
pragma solidity ^0.5.3;

contract HelloWorld {
  string public greet = "Hello World!";
}
</code></pre>
`

export default html
