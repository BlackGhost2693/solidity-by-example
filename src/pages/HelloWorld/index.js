import React, { Component } from 'react'
import Example from '../../components/Example'
import html from './index.html.js'

class HelloWorld extends Component {
  render() {
    return (
      <Example
        title="Hello World"
        html={html}
      />
    )
  }
}

export default HelloWorld
