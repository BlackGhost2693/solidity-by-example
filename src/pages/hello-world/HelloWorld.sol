// SPDX-License-Identifier: MIT
// compiler version must be greater than or equal to 0.7.6 and less than 0.9.0
pragma solidity ^0.7.6;

contract HelloWorld {
    string public greet = "Hello World!";
}
