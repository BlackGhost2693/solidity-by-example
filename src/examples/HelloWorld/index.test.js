import React from "react"
import { shallow } from "enzyme"
import HelloWorld from "./index"

test("it renders", () => {
  const component = shallow(<HelloWorld />)

  expect(component).toMatchSnapshot()
})
