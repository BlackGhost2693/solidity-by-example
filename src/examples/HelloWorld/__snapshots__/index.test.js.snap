// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<Example
  html="<p><code>pragma</code> specifies the compiler version of Solidity.</p>
<pre><code class=\\"language-solidity\\">/*
This code will not compile with a compiler earlier than version 0.5.1
and it will also not work on a compiler starting from version 0.6.0
*/
pragma solidity ^0.5.1;

contract HelloWorld {
    function hello() public returns (string memory) {
        return \\"Hello World!\\";
    }
}
</code></pre>
"
  title="Hello World"
/>
`;
